src/Conversion/ONNXToKrnl/Tensor/Pad.cpp

```diff
@@ -71,8 +72,15 @@ struct ONNXPadOpLowering : public ConversionPattern {
       if (constantValue.getType().isa<NoneType>()) {
         // Default to 0 if constant_value is not specified.
         cValue = create.math.constant(resElementType, 0);
-      } else
-        cValue = create.krnl.load(constantValue, {});
+      } else {
+        auto cst = llvm::dyn_cast<MemRefType>(constantValue.getType());
+        if(cst && cst.getRank() == 1) {
```

Comment:
Could you add a comment here why those steps are necessary?

Reply:
Right now, I assume that the input, if not a `tensor<f32>` will be `<1xf32>` and hardcoded the `load` to look at the the index `0`. It might be incorrect since all tensors are allowed but I have yet to see another kind of tensor. I think it will bring some discussion for sure but it should work for us at least.